src/harness/util.py

```diff
@@ -279,3 +279,129 @@ def convertRequestToRequestWithCpiSignature(private_key, cpi_id,
   request['cpiSignatureData']['protectedHeader'] = jwt_message[0]
   request['cpiSignatureData']['encodedCpiSignedData'] = jwt_message[1]
   request['cpiSignatureData']['digitalSignature'] = jwt_message[2]
+
+def getPPinPPA(boundarysPoints, ARCSEC):
+    import numpy as np
+    from shapely import geometry
+    
+    poly = geometry.Polygon([[lon,lat] for lon,lat in boundarysPoints])
+    arcsec = ARCSEC/3600.0
+    maxlat = -90 
+    minlat = 90
+    maxlon = -180
+    minlon = 180
+       
+    for point in boundarysPoints:
+        if point[1] > maxlat:
+            maxlat = point[1]
+        if point[1] < minlat:
+            minlat = point[1]         
+        if point[0] > maxlon:
+            maxlon = point[0]
+        if point[0] < minlon:
+            minlon = point[0]
+    
+    iLat = np.floor(minlat)
+    iLon = np.floor(minlon)
+    nL = np.floor((minlat - iLat)/arcsec)
+    nH = np.ceil((maxlat - iLat)/arcsec)
+    mL = np.floor((minlon - iLon)/arcsec)
+    mH = np.ceil((maxlon - iLon)/arcsec)
+        
+    latgrid = np.arange(iLat + nL*arcsec, iLat + nH *arcsec, arcsec)
+    longrid = np.arange(iLon + mL * arcsec, iLon + mH *arcsec, arcsec)
+    
+    if len(latgrid) > 0 and len(latgrid)>0:
```

Comment:
All the logic for rasterizing the PPA polygon can be made much more efficient by building one shapely MultiPoints from the grid location. And intersecting with the PPA Polygon/MultiPolygon, which will return a subset of the points.
This way all the heavy calculation is done internally in one pass of the C 'within' code within the shapely library.

However it seems that you only try to build a single point PPA here, and will return an error message if more than one point. In that case, your code wont suffer much performance penalty for that very specific case.

If we want to be more generic and support full real PPA zone (for some of the other test as well),  we could use a routine similar to the following for much more speedy rasterization of the PPA. Although I also think such routine should be part of the PPA reference model module.
```
def RasterizePolygon(polygon, resolution):
  """Rasterizes a shapely Polygon or MultiPolygon
   Inputs:
      polygon: A shapely polygon or multipolygon
      resolution: Gridding resolution (in degrees)
   Returns:
      a list of (lon,lat) points.
  """
  bounds = polygon.bounds
  lat_min = int(bounds[0] / resolution) * resolution
  lng_min = int(bounds[1] / resolution) * resolution
  lat_max = int(bounds[2] / resolution + 1) * resolution
  lng_max = int(bounds[3] / resolution + 1) * resolution
  mesh_x, mesh_y = np.mgrid[lat_min:lat_max:resolution,
                                                lng_min:lng_max:resolution]
  points = np.vstack((mesh_x.ravel(), mesh_y.ravel())).T
  pts = polygon.intersection(geometry.asMultiPoint(points))
  return [(pt.x, pt.y) for pt in pts]   # Build here the appropriate return format
```